agent/main.py

```
@@ -165,15 +171,68 @@ def main():
+                # 宿主 UI 已消失。此刻 msg_thread 仍永久阻塞在 recv() 上
+                # (Transceiver 的 timeout_ 是 milliseconds::max())，调 shut_down()
+                # 会挂死在它内部的 join() 里，所以只能硬退出。
+                gone = watchdog.exited_target or f"pid={watchdog.ppid}"
+                mfaalog.error(f"[Agent] 宿主进程 {gone} 已退出，Agent 立即终止")
+                watchdog.close()
+                cleanup_socket_file(socket_id)
+                sys.stdout.flush()
+                sys.stderr.flush()
+                os._exit(1)
+    except KeyboardInterrupt:
+        mfaalog.info("[Agent] 收到中断信号，正在退出")
```

Ruff (0.16.1)

[warning] String contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF001)